nixos/modules/virtualisation/containers.nix

```diff
@@ -714,7 +724,10 @@ in
             ${optionalString cfg.autoStart ''
               AUTO_START=1
             ''}
-            EXTRA_NSPAWN_FLAGS="${mkBindFlags cfg.bindMounts}"
+            EXTRA_NSPAWN_FLAGS="${(mkBindFlags cfg.bindMounts) +
+              (if cfg.extraFlags != null && cfg.extraFlags != []
+               then '' ${" " + concatStringsSep " " cfg.extraFlags}''
```

Comment:
Also, little nitpicks:
- you can use lib.optionalString instead of if-then-else
- you add one space before antiquotation, and one inside. One of them seems to be redundant

Reply:
@danbst  thank you for your suggestions to clean up the code